app/domain/authentication/resource_restrictions/extract_resource_restrictions.rb

```diff
@@ -6,8 +6,8 @@ module ResourceRestrictions
     ExtractResourceRestrictions = CommandClass.new(
       dependencies: {
         resource_restrictions_class: Authentication::ResourceRestrictions::ResourceRestrictions,
-        get_restriction_from_annotation: Authentication::ResourceRestrictions::GetRestrictionFromAnnotation,
-        fetch_resource_annotations_class: Authentication::ResourceRestrictions::FetchResourceAnnotations,
+        get_restriction_from_annotation: Authentication::ResourceRestrictions::GetRestrictionFromAnnotation.new,
+        fetch_resource_annotations_class: Authentication::ResourceRestrictions::FetchResourceAnnotations.new,
```

Comment:
you can remove the `_class` suffix

Reply:
Done but renamed first function in call to call_fetch_resource_annotations